### PR TITLE
Refactor/rename subpackages

### LINKED ISSUE #105: Rename subpackages
I propose the following API changes to subpackage names:

- `coxeter.shape_classes` ➡️ `coxeter.shapes`
- `coxeter.shape_families` ➡️ `coxeter.families`

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Rename shape_classes->shapes and shape_families->families.

## Motivation and Context

Resolves #105 

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [ ] New feature
- [x] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/coxeter/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/coxeter/blob/master/ContributorAgreement.md).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] I have updated the [changelog](https://github.com/glotzerlab/coxeter/blob/master/ChangeLog.txt) and the [credits](https://github.com/glotzerlab/coxeter/blob/master/doc/source/credits.rst).

### CODE REVIEW

File: coxeter/__init__.py [hunk 1]
```diff
@@ -13,9 +13,9 @@
 applications such as inertia tensors.
 """
 
-from . import shape_classes, shape_families
+from . import families, shapes
 from .shape_getters import from_gsd_type_shapes
 
-__all__ = ["shape_classes", "from_gsd_type_shapes", "shape_families"]
+__all__ = ["shapes", "from_gsd_type_shapes", "families"]
```
Comment (bdice) [inline coxeter/__init__.py:19]: Matches import order.
```suggestion
__all__ = ["families", "shapes", "from_gsd_type_shapes"]
```

File: coxeter/families/common_families.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
```
Comment (bdice) [inline coxeter/families/common_families.py:4]: Should this be renamed `common.py`?

File: coxeter/shapes/convex_spheropolygon.py [hunk 3]
```diff
@@ -61,7 +61,7 @@ def reorder_verts(self, clockwise=False, ref_index=0, increasing_length=True):
         """Sort the vertices.
 
         For more information see
-        :meth:`~coxeter.shape_classes._polygon.Polygon.reorder_verts`.
+        :meth:`~coxeter.shapes._polygon.Polygon.reorder_verts`.
```
Comment (bdice) [inline coxeter/shapes/convex_spheropolygon.py:64]: This is the preferred alias, right? (This also might just be a messed up reference that was supposed to say `convex_polygon`)
```suggestion
        :meth:`~coxeter.shapes.Polygon.reorder_verts`.
```

**Review by bdice — APPROVED** [review]
@vyasr I raised a few minor questions, fix up as needed and merge. LGTM.

File: coxeter/families/common_families.py (hunk 2 above)
Comment (bdice) [reply coxeter/families/common_families.py:4]: Likewise I wondered if `base_classes.py` should be `base.py` or `shape.py` or something like that.

Comment (vyasr) [reply coxeter/families/common_families.py:4]: I like that, yes I'll change it.

**Pull request merged**